auto_inertial_params/proposal.md

```diff
@@ -0,0 +1,157 @@
+
+```
+<model name="model_name">
+  <link/>
+</model>
+```
+
+## Motivation
+
+Currently, there are 2 major workflows used by the users to obtain the correct inertial parameters of their models:
+
+ * Using CAD software like [Fusion360](https://www.autodesk.in/products/fusion-360/overview?term=1-YEAR&tab=subscription) or [Solidworks](https://www.solidworks.com/). Many users design their robot models using such CAD software which provide plugins that automatically generate the URDF/SDF for their model. Such plugins handle the calculation of the inertial parameters. For example, Fusion360 provides the [Fusion2URDF](https://github.com/syuntoku14/fusion2urdf) plugin which automatically generates a URDF with all the inertial parameters.
+
+ * Another way is to use 3rd-party Mesh Processing Software like [Meshlab](https://www.meshlab.net/). Such softwares take the mesh file as an input and provide the inertial parameters as an output which can then be copied and pasted into the URDF/SDF file. This is also the method that was suggested in official [Classic Gazebo docs](https://classic.gazebosim.org/tutorials?cat=build_robot&tut=inertia).
+
+Both of these ways create a dependency on external software and might be complicated for beginners. Native support for this feature directly into libsdformat would facilitate the effortless generation of accurate simulations.
+
+## User Perspective
+
+To specify the `<inertial>` element of a `<link>` in SDFormat, the user needs to add the `<mass>`, `<pose>`, and `<inertia>` tags. Here `<mass>` is the mass of the link and `<pose>` is the pose of the Centre of Mass with respect to the link frame. The `<inertia>` tag, on the other hand, needs to enclose the following 6 tags:
+
+ * `<ixx>` (Moment of Inertia)
+
+ * `<iyy>` (Moment of Inertia)
+
+ * `<izz>` (Moment of Inertia)
+
+ * `<ixy>` (Product of Inertia)
+ 
+ * `<ixz>` (Product of Inertia)
+ 
+ * `<izy>` (Product of Inertia)
+
+This proposal suggests the addition of the following in `SDFormat`: 
+
+ 1.  `//inertial/@auto` attribute that would tell `libsdformat` to calculate the Inertial values (Mass, Mass Matrix & Inertial Pose) automatically for the respective link. 
+
+ 2. `//link/collision/density` element for specifying the density of the collision geometry. This density value would be used to calculate the inertial parameters of the respective collision geometry. Adding this as part of the `<collision>` tag would allow a user to simulate links with different material types for different collisions. By default, the value of density would be set equal to that of water which is 1000 kg/m^3.  
```

Comment:
apologies for suggesting changes so late, but I think it would be nice to specify a single density in `//link/inertial/density` that would apply to all associated shapes. This is slightly less flexible, but it avoids duplicating the density values in each collision element

Reply:
Thank you for the suggestion @scpeters. 
This seems like a good option to have. However, given the remaining time frame before the code freeze and the remaining work, I am not sure if it would be possible to get this implemented. Would it suffice if we update the SDF spec to include the `//link/inertial/density` element before the code freeze and implement this functionality later in a minor release by overloading the [Collision::CalculateInertial()](https://github.com/jasmeet0915/sdformat/blob/5af78f710d38ed37ed7cc392a3a1a03b9e771fae/src/Collision.cc#L220) function to take the density value from Link as a parameter [when called](https://github.com/jasmeet0915/sdformat/blob/5af78f710d38ed37ed7cc392a3a1a03b9e771fae/src/Link.cc#L614)?

We would probably need this for another element (`//collision/auto_inertia_params`) that was added in https://github.com/gazebosim/sdformat/pull/1304 (Apologies for not mentioning this element in the proposal)

Reply:
yes, that sounds totally reasonable. I think adding `//link/inertial/density` to the spec is enough for the first release